### PR TITLE
Remove more code for legacy Python

### PR DESCRIPTION
**Pull request by cclauss**
Use `flake8 --select=F821` to find and fix all `undefined names` in the codebase.
The majority of these are about builtins that were removed or moved in the transition from Python 2 to Python 3.

The failing test seems unrelated.

### CODE REVIEW

File: buildconfig/config_emsdk.py [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-from glob import glob
-import platform
 import logging
+import os
+import sys
+from glob import glob
+
 from distutils.sysconfig import get_python_inc
 
+from .config_unix import get_porttime_dep
```
Comment (ankith26) [inline buildconfig/config_emsdk.py:10]: why is this extra import here?

File: buildconfig/config_msys2.py [hunk 2]
```diff
@@ -16,6 +16,11 @@
 from glob import glob
 from distutils.sysconfig import get_python_inc
 
+try:
+    BuildError
+except NameError:
+    BuildError = TypeError
```
Comment (ankith26) [inline buildconfig/config_msys2.py:22]: what is this for? Why is this here and in some other files?

File: buildconfig/stubs/pygame/.flake8 [hunk 3]
```diff
@@ -16,10 +16,11 @@
 #       W504 line break after binary operator
 
 [flake8]
-ignore = F401, F403, F405, F811, E301, E302, E305, E501, E701, E704, E741, B303, W504
+ignore = B303, E301, E302, E305, E501, E701, E704, E741, F401, F403, F405, F811, W504
 # We are checking with Python 3 but many of the stubs are Python 2 stubs.
 # A nice future improvement would be to provide separate .flake8
 # configurations for Python 2 and Python 3 files.
```
Comment (ankith26) [inline buildconfig/stubs/pygame/.flake8:22]: these comments are rather outdated, this PR would be a good opportunity to clean up these :smile:

**Review by ankith26 — COMMENTED** [review]
Nice cleanups, thanks a ton!

Left a few reviews to address before approval

File: buildconfig/config_emsdk.py [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
+
 configcommand = os.environ.get('SDL_CONFIG', 'sdl-config',)
 configcommand = configcommand + ' --version --cflags --libs'
 localbase = os.environ.get('LOCALBASE', '')
-if os.environ.get('PYGAME_EXTRA_BASE', ''):
-    extrabases = os.environ['PYGAME_EXTRA_BASE'].split(':')
-else:
-    extrabases = []
+extrabases = os.environ.get('PYGAME_EXTRA_BASE', '').split(':')
```
[the remark this answers is not in the document]
Comment (cclauss) [reply buildconfig/config_emsdk.py:15]:
```suggestion
if os.environ.get('PYGAME_EXTRA_BASE'):
    extrabases = os.environ['PYGAME_EXTRA_BASE'].split(':')
else:
    extrabases = []
```

File: buildconfig/config_msys2.py (hunk 2 above)
Comment (cclauss) [reply buildconfig/config_msys2.py:22]: It is an _undefined name_ which has the potential to raise `NameError` at runtime so let's avoid that.

File: buildconfig/stubs/pygame/.flake8 (hunk 3 above)
Comment (cclauss) [inline buildconfig/stubs/pygame/.flake8:22]:
```suggestion
```

File: buildconfig/stubs/pygame/.flake8 [hunk 5]
```diff
@@ -16,10 +16,11 @@
...
-builtins = StandardError,apply,basestring,buffer,cmp,coerce,execfile,file,intern,long,raw_input,reduce,reload,unichr,unicode,xrange
-exclude = .venv*,@*,.git
+# builtins = StandardError,apply,basestring,buffer,cmp,coerce,execfile,file,intern,long,raw_input,reduce,reload,unichr,unicode,xrange
```
Comment (cclauss) [inline buildconfig/stubs/pygame/.flake8:23]:
```suggestion
```

File: buildconfig/config_emsdk.py (hunk 1 above)
Comment (cclauss) [reply buildconfig/config_emsdk.py:10]: It is an _undefined name_ that can potentially raise `NameError` at runtime so let's avoid that.

Comment (ankith26) [reply buildconfig/config_emsdk.py:10]: That condition that calls `get_porttime_dep` in this file does not actually execute at runtime... I think that codepath should be removed instead of adding this import

File: buildconfig/config_msys2.py (hunk 2 above)
Comment (ankith26) [reply buildconfig/config_msys2.py:22]: Again... the codepath that actually calls `BuildError` is super unlikely to run, and even if it does, I think it's better if it errored with `RuntimeError` directly in code instead of dealing with `BuildError` aliases which is not even present elsewhere in the code

**Comment by cclauss** [comment]
https://flake8.pycqa.org/en/latest/user/error-codes.html

On the flake8 test selection, this PR does _not_ focus on "_style violations_" (the majority of flake8 error codes that [__psf/black__](https://github.com/psf/black) can autocorrect).  Instead, these tests are focus on runtime safety and correctness:
* E9 tests are about Python syntax errors usually raised because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python, they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).  Python >= 3.8 will raise SyntaxWarnings on these instances.
* F7 tests logic errors and syntax errors in type hints
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python, a __NameError__ is raised which will halt/crash the script on the user.

File: buildconfig/config_emsdk.py (hunk 1 above)
Comment (cclauss) [reply buildconfig/config_emsdk.py:10]: I feel unqualified to make this change.

File: buildconfig/config_msys2.py (hunk 2 above)
Comment (ankith26) [reply buildconfig/config_msys2.py]: Nice, I see you fixed this. But you are raising `TypeError` there which seems to be the wrong error to raise here, ideally it should be a `ValueError` (because the value should have been 32 or 64) or a `RuntimeError`

File: buildconfig/config_emsdk.py (hunk 1 above)
Comment (ankith26) [reply buildconfig/config_emsdk.py:10]: There is a `if not is_wasm:` block guarding the `get_porttime_dep` function call... but at the top of the file there is also a systemexit happening for the same condition, so this can be removed safely.

If you wish, I could make a commit for it in this branch, or you could do it yourself too :)

Comment (ankith26) [reply buildconfig/config_emsdk.py:8]: You removed that unused block correctly, but the import is still here so this needs to be removed too

File: buildconfig/config_emsdk.py [hunk 6]
```diff
@@ -1,27 +1,26 @@
...
+import sys
+from glob import glob
+
+from .config_unix import get_porttime_dep
 
 configcommand = os.environ.get('SDL_CONFIG', 'sdl-config',)
 configcommand = configcommand + ' --version --cflags --libs'
 localbase = os.environ.get('LOCALBASE', '')
-if os.environ.get('PYGAME_EXTRA_BASE', ''):
+if os.environ.get('PYGAME_EXTRA_BASE'):
```
Comment (ankith26) [inline buildconfig/config_emsdk.py:13]: minor nitpick
```suggestion
if "PYGAME_EXTRA_BASE" in os.environ:
```

**Review by ankith26 — APPROVED** [review]
LGTM! 🎉 
Thanks again for the PR

**Review by dr0id — APPROVED** [review]

**Review by illume — APPROVED** [review]
👍 🎉 thanks
